### PR TITLE
Externals: Add Dear Implot 

### PR DESCRIPTION
**Pull request by Sam-Belliveau**
This is a dependency for the branches I am working on. I am currently blocked on this.

I do not know if this builds on windows.

### CODE REVIEW

**Comment by delroth** [comment]
@dolphin-emu-bot rebuild

**Comment by delroth** [comment]
@dolphin-emu-bot rebuild

**Comment by iwubcode** [comment]
This should be using a submodule for the `implot` code.  See my [SPIRVCross](https://github.com/dolphin-emu/dolphin/tree/master/Externals/spirv_cross) implementation for an example.

File: Source/Core/VideoCommon/CMakeLists.txt [hunk 1]
```diff
@@ -168,6 +168,7 @@ PRIVATE
   spng
   xxhash
   imgui
+  implot
```
Comment (iwubcode) [inline Source/Core/VideoCommon/CMakeLists.txt:171]: Changes to add `implot` as a dependency should be done in the PR that is actually using them

**Comment by Sam-Belliveau** [comment]
> This should be using a submodule for the `implot` code. See my [SPIRVCross](https://github.com/dolphin-emu/dolphin/tree/master/Externals/spirv_cross) implementation for an example.

is there a reason that ImGui is not done like this? I just wanted to keep it consistent with ImGui.

**Comment by iwubcode** [comment]
> is there a reason that ImGui is not done like this? I just wanted to keep it consistent with ImGui.

@Sam-Belliveau  - it is an older dependency

**Review by iwubcode — APPROVED** [review]
Looks ok to me.  Thanks!

**Comment by BhaaLseN** [comment]
Looks good in general, but you might wanna clean up those commits (because many of them have no discernable meaning when looked at in isolation).

-----

To squash those commits into one before continuing; you can use the following commands (replace `upstream` with the name of the Dolphin remote and `origin` with the name of your Fork remote. Also, make sure you are on your branch when you do this):
```
git fetch upstream
git reset --soft upstream/master
git commit -m "your commit message here"
git push -f origin HEAD
```
The first command makes sure you get the latest commits from `master` (from the Dolphin repository) to work on; the second picks all your changes, switches to that `master` state and puts the changes in the staging area. The third command then commits it so you have one commit total that has all the changes (which you can also replace with your commit method of choice, such as `git gui` or any other GUI tool); and the forth command pushes your current branch into your own remote (which should be your Fork) while overwriting all previous changes (to update this PR).

**Comment by Sam-Belliveau** [comment]
> Looks good in general, but you might wanna clean up those commits (because many of them have no discernable meaning when looked at in isolation).
> 
> To squash those commits into one before continuing; you can use the following commands (replace `upstream` with the name of the Dolphin remote and `origin` with the name of your Fork remote. Also, make sure you are on your branch when you do this):
> 
> ```
> git fetch upstream
> git reset --soft upstream/master
> git commit -m "your commit message here"
> git push -f origin HEAD
> ```
> 
> The first command makes sure you get the latest commits from `master` (from the Dolphin repository) to work on; the second picks all your changes, switches to that `master` state and puts the changes in the staging area. The third command then commits it so you have one commit total that has all the changes (which you can also replace with your commit method of choice, such as `git gui` or any other GUI tool); and the forth command pushes your current branch into your own remote (which should be your Fork) while overwriting all previous changes (to update this PR).

Thank you I really could not remember how to do this

**Pull request merged**